processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -32,13 +32,13 @@
  * provided to the Aggregator through the MetricSelector object, so whatever creates that object gets to choose how
  * the data is actually stored and accessed.
  */
-public interface AggregatorFactory
+public abstract class AggregatorFactory
```

Comment:
I think given that switching this to abstract class will be source-incompatible anyway, I have a general preference for an interface `Blah` + an abstract `AbstractBlah` or `BaseBlah`, since I think it makes life a little simpler for implementors that may not want to be bothered by what exactly the abstract base class is doing.

Reply:
#5475